### PR TITLE
Darwin fixes

### PR DESCRIPTION
**Pull request by toonn**
Hi, I'm working on fixing the FFmpeg 7 build in Nixpkgs on Darwin/macOS and I figured some of the changes may be welcome upstream.

[Using `CMAKE_INSTALL_FULL_*` paths](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) will avoid a problem due to us setting `CMAKE_INSTALL_*` to absolute paths, which causes the manual prefixing currently done to create non-existing paths.

The CMakeLists.txt changes are due to us using Clang on Darwin.

The change in xevdm_util.c might be a bug or I might be wrong and it should indeed iterate in steps of size 2.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -78,7 +78,7 @@ if(XEVD_APP_STATIC_BUILD)
 endif(XEVD_APP_STATIC_BUILD)
 
 cmake_policy(SET CMP0048 NEW)
-set(CMAKE_C_STANDARD 99)
```
Comment (dariusz-f) [inline CMakeLists.txt:81]: For now we want to keep C99

**Review by dariusz-f — CHANGES_REQUESTED** [review]
Please rebase this PR and keep C99 standard.

File: CMakeLists.txt (hunk 1 above)
Comment (toonn) [reply CMakeLists.txt:81]: I'm not sure why Github isn't updating this PR. I rebased and force-pushed. This should be at 2ebfd0794b92c5f5a2bf9b1220969dca509176fb.

Comment (toonn) [reply CMakeLists.txt:81]: Welp, I'm an idiot. Confusing this with the PR on xeve. I'll give this one the same treatment tomorrow.

Comment (toonn) [reply CMakeLists.txt]: Rebased and dropped everything that's not strictly related to Darwin, opened separate PRs for those changes.

**Review by dariusz-f — APPROVED** [review]

**Pull request merged**